File: Backend/src/main/java/com/shopsphere/backend/service/WishlistService.java
package com.shopsphere.backend.service;

import com.shopsphere.backend.dto.wishlist.WishlistResponse;

import java.util.List;

public interface WishlistService {

    WishlistResponse addToWishlist(Long productId);

    List<WishlistResponse> getMyWishlist();

    void removeFromWishlist(Long productId);
}

File: shopsphere-frontend/src/services/wishlistService.js
import axiosInstance from "./axiosInstance";

export const getMyWishlist = () => {
  return axiosInstance.get("/wishlist");
};

export const addToWishlist = (productId) => {
  return axiosInstance.post(
    `/wishlist/${productId}`
  );
};

export const removeFromWishlist = (productId) => {
  return axiosInstance.delete(
    `/wishlist/${productId}`
  );
};